src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -564,31 +594,48 @@ def launch_fluent(
             raise TypeError(
                 f"launch_fluent() got an unexpected keyword argument {next(iter(kwargs))}"
             )
-
     del kwargs
     argvals = locals()
 
-    if start_watchdog is None and not show_gui:
-        start_watchdog = True
-
-    new_session, meshing_mode, argvals, mode = _get_session_info(argvals, mode)
-    _raise_exception_g_gu_in_windows_os(additional_arguments)
-
     if pypim.is_configured():
         fluent_launch_mode = LaunchMode.PIM
-    elif os.getenv("PYFLUENT_LAUNCH_CONTAINER") == "1":
-        fluent_launch_mode = LaunchMode.CONTAINER
+    elif start_container is True or (
+        start_container is None
+        and container_dict
+        or os.getenv("PYFLUENT_LAUNCH_CONTAINER") == "1"
+    ):
+        if check_docker_support():
+            start_container = True
```

Comment:
rather than assigning `start_container` at this point, we could just `del` it and refer only to `fluent_launch_mode` hereafter?